<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MODI_FIRMWARE_UPDATER</class>
 <widget class="QDialog" name="MODI_FIRMWARE_UPDATER">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>786</width>
    <height>331</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="2" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="local_module">
       <property name="text">
        <string>Current module: ?? (??) </string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="curr_module_img">
       <property name="maximumSize">
        <size>
         <width>64</width>
         <height>64</height>
        </size>
       </property>
       <property name="font">
        <font>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="pixmap">
        <pixmap>assets/imgs/network.png</pixmap>
       </property>
       <property name="scaledContents">
        <bool>true</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="local_percentage">
       <property name="text">
        <string>?? %</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QProgressBar" name="push_bar">
       <property name="value">
        <number>0</number>
       </property>
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="push_button">
       <property name="text">
        <string>Tap multiple times to 
speed up the update</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="update_network_esp32">
       <property name="text">
        <string>Update Network Module</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>assets/imgs/network.png</normaloff>assets/imgs/network.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>64</width>
         <height>64</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="update_stm32_modules">
       <property name="text">
        <string>Update STM32 Modules</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>assets/imgs/button.png</normaloff>assets/imgs/button.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>64</width>
         <height>64</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="update_network_stm32">
       <property name="text">
        <string>Update Network Base</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>assets/imgs/network.png</normaloff>assets/imgs/network.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>64</width>
         <height>64</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="status_label">
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <weight>75</weight>
       <italic>false</italic>
       <bold>true</bold>
       <underline>false</underline>
       <kerning>true</kerning>
      </font>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="frameShape">
      <enum>QFrame::WinPanel</enum>
     </property>
     <property name="text">
      <string>Welcome to MODI Firmware Updater! 
 This GUI application utilizes PyMODI as its backend :) 

 To start updating your MODI firmware, click an appropriate button below!</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
 <slots>
  <slot>slot1()</slot>
  <slot>slot2()</slot>
  <slot>slot3()</slot>
 </slots>
</ui>
